plugins/inputs/suricata/suricata.go

```diff
@@ -0,0 +1,229 @@
+package suricata
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	// InBufSize is the input buffer size for JSON received via socket.
+	// Set to 10MB, as depending on the number of threads the output might be
+	// large.
+	InBufSize = 10 * 1024 * 1024
+)
+
+// Suricata is a Telegraf input plugin for Suricata runtime statistics.
+type Suricata struct {
+	sync.Mutex
+
+	Source    string `toml:"source"`
+	Delimiter string `toml:"delimiter"`
+
+	InputListener *net.UnixListener
+	Ctx           context.Context
```

Comment:
Don't save this Context on the struct, just pass it down.  This matches the recommendations in the context package:
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:

Reply:
Done.